dev-clean/src/main/scala/sai/lang/LLVM.scala

```diff
@@ -799,6 +802,15 @@ class MyVisitor extends LLVMParserBaseVisitor[LAST] {
   override def visitFloatConst(ctx: LLVMParser.FloatConstContext): LAST = {
     val floatStr = ctx.FLOAT_LIT.getText
     if (floatStr.contains('.')) FloatConst(floatStr.toFloat)
+    else if (floatStr.startsWith("0xK")) {
+      // x86_fp80
+      val numStr = floatStr.substring(3)
```

Comment:
maybe put this part and the toString function into codegen?

Reply:
So change the interface of `FloatLitConst` such that it takes a single string instead of a list of string?

Reply:
yes, since the list of strings (as well as its toString implementation) is a specific backend representation, which should not be propagated into the front-end parser.